app/src/main/java/org/mozilla/fenix/settings/advanced/LocaleManagerExtension.kt

```diff
@@ -0,0 +1,72 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.advanced
+
+import android.content.Context
+import android.content.res.Resources
+import android.os.Build
+import mozilla.components.support.locale.LocaleManager
+import java.util.Locale
+
+/**
+ * Returns a list of currently supported locales, with the system default set as the first one
+ */
+fun LocaleManager.getSupportedLocales(): List<Locale> {
+    val localeList: MutableList<Locale> = ArrayList()
+    localeList.add(getSystemDefault())
+    localeList.add(Locale("eu"))
+    localeList.add(Locale("ca"))
+    localeList.add(Locale("zh", "CN"))
+    localeList.add(Locale("zh", "TW"))
+    localeList.add(Locale("cs"))
+    localeList.add(Locale("nl"))
+    localeList.add(Locale("en"))
+    localeList.add(Locale("fi"))
+    localeList.add(Locale("fr"))
+    localeList.add(Locale("de"))
+    localeList.add(Locale("hu"))
+    localeList.add(Locale("in"))
+    localeList.add(Locale("it"))
+    localeList.add(Locale("ja"))
+    localeList.add(Locale("ko"))
+    localeList.add(Locale("pl"))
+    localeList.add(Locale("pa", "IN"))
+    localeList.add(Locale("ru"))
+    localeList.add(Locale("sk"))
+    localeList.add(Locale("es"))
+    localeList.add(Locale("es", "AR"))
+    localeList.add(Locale("es", "ES"))
+    localeList.add(Locale("sv", "SE"))
+    localeList.add(Locale("ta"))
+    localeList.add(Locale("tr"))
+    return localeList
+}
+
+/**
+ * Returns the locale that corresponds to the language stored locally by us. If no suitable one is found,
+ * return default.
+ */
+fun LocaleManager.getSelectedLocale(context: Context, localeList: List<Locale>): Locale {
```

Comment:
`localeList` definitely makes this easier to test, but it doesn't seem to me that consumers should have to pass that in order to get the selected locale.  What about making it an optional param?  `localeList: List<Locale> = getSupportedLocales()`

Reply:
Yep, sounds right, they shouldn't. Added as optional